opentelemetry-sdk/src/metrics/data/mod.rs

```diff
@@ -275,6 +275,16 @@ impl<T> HistogramDataPoint<T> {
     pub fn exemplars(&self) -> impl Iterator<Item = &Exemplar<T>> {
         self.exemplars.iter()
     }
+
+    /// Returns an iterator over the bucket boundaries in [HistogramDataPoint].
+    pub fn bounds(&self) -> impl Iterator<Item = &f64> {
```

Comment:
We could make the API more ergonomic by returning `impl Iterator<Item = f64>` instead of `impl Iterator<Item = &f64>`:

```rust
/// Returns an iterator over the bucket boundaries in [HistogramDataPoint].
pub fn bounds(&self) -> impl Iterator<Item = f64> + '_ {
    self.bounds.iter().copied()
}

/// Returns an iterator over the bucket counts in [HistogramDataPoint].
pub fn bucket_counts(&self) -> impl Iterator<Item = u64> + '_ {
    self.bucket_counts.iter().copied()
}
```

Reply:
yes makes sense. Made this change and updated exporters accordingly.